api/core/helper/ssl_verification.py

```diff
@@ -0,0 +1,45 @@
+"""Helpers for resolving SSL verification settings."""
+
+import ssl
+from functools import lru_cache
+from typing import Any, Union
+
+import certifi
+
+SslVerifySetting = Union[bool, ssl.SSLContext]
+
+
+@lru_cache(maxsize=1)
+def _cached_ssl_context() -> ssl.SSLContext:
+    """Return a cached default SSL context backed by certifi's CA bundle."""
+    return ssl.create_default_context(cafile=certifi.where())
+
+
+def get_cached_ssl_context() -> ssl.SSLContext:
+    """Expose the cached SSL context so callers reuse the same instance."""
+    return _cached_ssl_context()
+
+
+def parse_ssl_verify_config(value: Any, *, config_name: str) -> SslVerifySetting:
+    """Parse a configuration value into an httpx-compatible verify setting.
+
+    Accepts booleans, strings, or ssl.SSLContext instances. Truthy strings resolve
+    to the cached SSL context while falsy strings disable verification.
+    """
+
+    if isinstance(value, ssl.SSLContext):
+        return value
+
+    if value is None:
+        return get_cached_ssl_context()
+
+    if isinstance(value, bool):
+        return get_cached_ssl_context() if value else False
```

Comment:
Try to prefer splitting into named parameters for this method. Prevent mixing up the value in different types and usages.